src/components/focusManager.js

```diff
@@ -131,6 +146,18 @@ define(['dom'], function (dom) {
             }
         }
 
+        var elementRect = getOffset(elem);
+
+        // not currently visible
+        if (!elementRect.width && !elementRect.height) {
+            return false;
+        }
+
+        // out of view
+        if (viewport !== false && (elementRect.right < 0 || elementRect.bottom < 0 || elementRect.right >= window.innerWidth || elementRect.top >= window.innerHeight)) {
```

Comment:
By default it should test viewport boundaries (means 'not explicitly false'), so we cannot focus on element out of view (not-seen), otherwise the screen will jump around. But if we need to focus on element out of view (for example scroll up to default button) then caller pass 'false' explicitly. I used this once at viewManager.onViewChange.

Reply:
This doesn't sound like a good variable name and default value to me... a default value for omitted variable is `undefined` which is falsy, so I propose to name it `forceScroll` and set it to `true` when needed